.github/workflows/cd.yml

```diff
@@ -0,0 +1,109 @@
+name: Deploy
+
+on:
+  push:
+    tags:
+    - '*.*.*'
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+
+    strategy:
+      matrix:
+        python-version: [2.7, 3.8]
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Setup Python 2
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Ensure pip version for python 2
+      run: |
+        if [ "${{ matrix.python-version}}" == "2.7" ] ; then
+            echo "pip_v=pip < 21.0" >> $GITHUB_ENV
+        else
+            echo "pip_v=pip" >> $GITHUB_ENV
+        fi
+
+    - name: Build
+      run: |
+        python -m pip install --upgrade "${{ env.pip_v }}" setuptools wheel
+        python setup.py sdist bdist_wheel
+
+    - name: Upload artifacts
+      uses: actions/upload-artifact@v2
+      with:
+        name: distfiles_${{ github.run_id }}
+        path: dist
+
+  release:
+    needs: ["build"]
+    runs-on: ubuntu-20.04
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Setup Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
+
+    - name: Get tag and tracker versions
+      id: version
+      run: |
+        echo ::set-output name=TAG_VERSION::${GITHUB_REF#refs/*/}
+        echo "##[set-output name=PYTHON_TRACKER_VERSION;]$(python setup.py --version)"
+
+    - name: Fail if version mismatch
+      if: ${{ steps.version.outputs.TAG_VERSION != steps.version.outputs.PYTHON_TRACKER_VERSION }}
+      run: |
+        echo "Tag version (${{ steps.version.outputs.TAG_VERSION }}) doesn't match version in project (${{ steps.version.outputs.PYTHON_TRACKER_VERSION }})"
+        exit 1
+
+    - name: Release
+      uses: softprops/action-gh-release@v1
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      with:
+        tag_name: ${{ github.ref }}
+        name: Snowplow Python Tracker v${{ steps.version.outputs.PYTHON_TRACKER_VERSION }}
+        draft: false
+        prerelease: ${{ contains(steps.version.outputs.TAG_VERSION, 'rc') }}
+
+  publish:
```

Comment:
Should we publish before we release to Github instead? I think the publish step is more likely to fail but we'd have released to Github already once it had failed.
If we do this, we should also move the Version check (perhaps even before we build it, or to the beginning of publish if thats preferred?)

Reply:
Thank you for pointing this out. Indeed, the release is less likely to fail. And the Version-check also makes sense to move ahead to fail fast if there is mismatch.